### DISCUSSION TITLE
Support for Cosy tariff?

### DISCUSSION THREAD

**Discussion by dpritt** [BottlecapDave/HomeAssistant-OctopusEnergy#608] [Ideas]
I am on the Cosy tariff, which has a two 'cheaper' periods, a 'high' period between 16:00 and 19:00 and the rest of the time a standard price. 

How can I get this to be recognised, and obtain the relevant periods' consumption and cost. Currently I only get peak and off-peak value, which are wrong.

**Comment by BottlecapDave** [comment]
Hello. The integration goes for broad support across all tariffs, and only goes deeper where support can not be achieved via other HA methods (e.g. intelligent). I don't have the time/resource to be inline with a massive energy supplier.

In the case of cosy, all data is available, rate sensors work as designed and the off peak sensor is compatible (as this sensor is applicable in other tariffs). I assume you're issue is referring to the accumulative consumption/cost sensors? These sensors are again generic and so the standard peak (which would be peak in other tariffs as it's not off peak) and "super peak" are lumped together.

What is your desired goal?

**Comment by dpritt** [comment]
Dave,
 
Thanks for your response.
 
Yes it is the accumulative entities I would like to use to build up my month-to-date use and costs across each of the tariff time slots. 
 
I could try to push the half-hour values into a SQL table and write some code to group the relevant half-hour periods to produce the day and month-to-date values, but I am trying to avoid building anything outside the Home Assistant package.
 
I am impressed by what you have written, I tried this myself when I was on Agile, but your integration is a lot neater and more powerful.
 
Regards
 
Dave
